test/JDBC/input/functions/BABEL-3953-datetrunc-vu-prepare.sql

```diff
@@ -0,0 +1,233 @@
+-- Test with date datatype
+CREATE VIEW DATETRUNC_vu_prepare_v1 AS (
+    select 
+        datetrunc(year, cast('2020-04-15' as date)) as dt1,
+        datetrunc(yy, cast('2020-04-15' as date)) as dt2,
+        datetrunc(yyyy, cast('2020-04-15' as date)) as dt3,
+        datetrunc(quarter, cast('2020-04-15' as date)) as dt4,
+        datetrunc(qq, cast('2020-04-15' as date)) as dt5,
+        datetrunc(q, cast('2020-04-15' as date)) as dt6,
+        datetrunc(month, cast('2020-04-15' as date)) as dt7,
+        datetrunc(mm, cast('2020-04-15' as date)) as dt8,
+        datetrunc(m, cast('2020-04-15' as date)) as dt9,
```

Comment:
Please add all invalid testcase like datetrunc(NULL,NULL),datetrunc('NULL',NULL) etc. and others with different data types.